File: SocialBackend/src/main/java/com/collaborate/SocialBackend/service/FriendService.java
package com.collaborate.SocialBackend.service;

import java.util.List;

import com.collaborate.SocialBackend.model.Friend;
import com.collaborate.SocialBackend.model.User;

public interface FriendService {
	List<User> listOfSuggestedUsers(String userName);
	void friendRequest(Friend friend);
	List<Friend> pendingRequests(String toID);
	void updatePendingRequests(Friend friend);
	List<String> listOfFriends(String userName);
	
}

File: SocialFrontend/WebContent/services/friendservice.js
/**
 * 
 */
app.factory('FriendService',function($http){
	var friendService={}
	var BASE_URL="http://localhost:8080/SocialMiddleware"
		friendService.listOfSuggestedUsers=function(){
		return $http.get(BASE_URL + "/getsuggestedusers")
	}
	friendService.sendFriendRequest=function(toId){
		return $http.get(BASE_URL +"/friendrequest/"+toId)
	}
	
	friendService.listOfSuggestedUsers=function(){
		return $http.get(BASE_URL +"/pendingrequests")
	}
	return friendService;
})